### PR TITLE
Configurable kafka protocol version for msg consuming by jaeger ingester

### PR DESCRIPTION
**Pull request by marqc**
## Which problem is this PR solving?

Jaeger ingester silently doesn't consume messages, when kafka server doesn't support kafka protocol in version 0.0.9.

## Short description of the changes

I added new `kafka.consumer.protocol-version` configuration option, that allows specify kafka protocol version to use. Default is "0.9.0.0" which was used by bsm/sarama-cluster as default version (for backwards compatibility).

### CODE REVIEW

File: pkg/kafka/consumer/config.go [hunk 1]
```diff
@@ -36,10 +37,11 @@ type Builder interface {
...
 type Configuration struct {
-	Brokers  []string
-	Topic    string
-	GroupID  string
-	ClientID string
+	Brokers              []string
+	Topic                string
+	GroupID              string
+	ClientID             string
+	KafkaProtocolVersion string
```
Comment (yurishkuro) [inline pkg/kafka/consumer/config.go]: it's not necessary to quality this with Kafka prefix - the whole thing is Kafka configuration.

File: cmd/ingester/app/flags.go [hunk 2]
```diff
@@ -63,6 +65,8 @@ const (
 	DefaultEncoding = kafka.EncodingProto
 	// DefaultDeadlockInterval is the default deadlock interval
 	DefaultDeadlockInterval = 1 * time.Minute
+	// DefaultKafkaProtocolVersion is the default kafka communication protocol version
+	DefaultKafkaProtocolVersion = "0.9.0.0"
```
Comment (yurishkuro) [inline cmd/ingester/app/flags.go]: why do we need to have explicit default? I'd rather leave it unset.

File: cmd/ingester/app/flags.go [hunk 3]
```diff
@@ -49,6 +49,8 @@ const (
 	SuffixParallelism = ".parallelism"
 	// SuffixHTTPPort is a suffix for the HTTP port
 	SuffixHTTPPort = ".http-port"
+	// SuffixKafkaProtocolVersion Kafka protocol version - must be supported by kafka server
+	SuffixKafkaProtocolVersion = ".protocol-version"
```
Comment (yurishkuro) [inline cmd/ingester/app/flags.go]: this should be grouped with Kafka-related field above, above `.encoding` 

Probably doesn't need Kafka in the name

File: cmd/ingester/app/builder/builder.go [hunk 4]
```diff
@@ -55,6 +55,7 @@ func CreateConsumer(logger *zap.Logger, metricsFactory metrics.Factory, spanWrit
 		Topic:                options.Topic,
 		GroupID:              options.GroupID,
 		ClientID:             options.ClientID,
+		Version:              options.Version,
```
Comment (yurishkuro) [inline cmd/ingester/app/builder/builder.go]: let's call this ProtocolVersion

Comment (marqc) [reply cmd/ingester/app/builder/builder.go]: renamed

**Comment by yurishkuro** [comment]
tests  are failing
```
--- FAIL: TestOptionsWithFlags (0.00s)
    flags_test.go:46: 
        	Error Trace:	flags_test.go:46
        	Error:      	Not equal: 
        	            	expected: "1.0.0"
        	            	actual  : ""
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-1.0.0
        	            	+
        	Test:       	TestOptionsWithFlags
    flags_test.go:47: 
        	Error Trace:	flags_test.go:47
        	Error:      	Not equal: 
        	            	expected: 5
        	            	actual  : 1000
        	Test:       	TestOptionsWithFlags
    flags_test.go:48: 
        	Error Trace:	flags_test.go:48
        	Error:      	Not equal: 
        	            	expected: 120000000000
        	            	actual  : 60000000000
        	Test:       	TestOptionsWithFlags
```

**Comment by marqc** [comment]
@yurishkuro I made a typo in test data config. Fixed now.

**Comment by yurishkuro** [comment]
Q: wouldn't a symmetric change in prroducer code make sense as well?

**Comment by marqc** [comment]
> Q: wouldn't a symmetric change in producer code make sense as well?

Yes. Good point. I can prepare some PR for cmd/collector pkg/kafka/producer tomorrow

**Pull request merged**